server/neptune/workflows/internal/terraform/workflow.go

```diff
@@ -0,0 +1,169 @@
+		}
+	}
+	// Wait for plan review signal
+	planReview := r.WaitUntilPlanIsReviewed(ctx)
+	if planReview.Status == Rejected {
+		return nil
+	}
+	// Run apply steps
+	for _, step := range r.request.Root.Plan.Steps {
+		err := r.runStep(ctx, step)
+		if err != nil {
+			return errors.Wrap(err, "running step")
+		}
+	}
+	// Cleanup
+	err = workflow.ExecuteActivity(ctx, r.Cleanup, activities.CleanupRequest{}).Get(ctx, nil)
+	if err != nil {
+		return errors.Wrap(err, "cleaning up")
+	}
+	return nil
+}
+
+// TODO: wrap each case statement's ExecuteActivity with a specific Runner implementation;
+// activity itself should just handle the non-deterministic code chunk (i.e. running terraform operation, state updates)
+// ex:
+//type Runner interface {
+//	Run(ctx workflow.Context, step deploy.Step) error
+//}
+//case "init":
+//	err = initStepRunner.Run(ctx, step)
+//	if err != nil {
+//		return errors.Wrap(err, "executing terraform init")
+//	}
+func (r *Runner) runStep(ctx workflow.Context, step Step) error {
+	var err error
+	switch step.StepName {
+	case "init":
+		err = workflow.ExecuteActivity(ctx, r.TerraformInit, activities.TerraformInitRequest{}).Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "executing terraform init")
+		}
+	case "plan":
+		err = workflow.ExecuteActivity(ctx, r.TerraformPlan, activities.TerraformPlanRequest{}).Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "executing terraform plan")
+		}
+		err = workflow.ExecuteActivity(ctx, r.Notify, activities.NotifyRequest{}).Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "notifying plan result")
+		}
+	case "apply":
+		err = workflow.ExecuteActivity(ctx, r.TerraformApply, activities.TerraformApplyRequest{}).Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "executing terraform apply")
+		}
+		err = workflow.ExecuteActivity(ctx, r.Notify, activities.NotifyRequest{}).Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "notifying apply result")
+		}
+	case "run":
+		err = workflow.ExecuteActivity(ctx, r.ExecuteCommand, activities.ExecuteCommandRequest{}).Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "executing custom command")
+		}
+	case "env":
+		err = workflow.ExecuteActivity(ctx, r.ExecuteCommand, activities.ExecuteCommandRequest{}).Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "exporting custom env variables")
+		}
+	}
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type PlanReview struct {
+	Status PlanStatus
+}
+
+//TODO: should we build a timer select branches to notify users that an approval is still needed/way to cancel long running exection?
+func (r *Runner) WaitUntilPlanIsReviewed(ctx workflow.Context) PlanReview {
+	s := workflow.NewSelector(ctx)
+	var planReview PlanReview
+	signalChan := workflow.GetSignalChannel(ctx, "planreview-repo-root")
+	s.AddReceive(signalChan, func(c workflow.ReceiveChannel, more bool) {
+		c.Receive(ctx, &planReview)
+	})
+	s.Select(ctx)
```

Comment:
btw you don't need a select for just one branch, you can just call the channel's receive.